pkg/awsManager/vpcVpnManager.go

```diff
@@ -0,0 +1,662 @@
+	var token *string
+	for {
+		vpcList, err := client.DescribeVpcs(&ec2.DescribeVpcsInput{NextToken: token})
+		if err != nil {
+			fmt.Println("Error:", err)
+			fmt.Println("Error: Cant list VPC ")
+			return nil, err
+		}
+
+		for _, vpcs := range vpcList.Vpcs {
+			if *vpcs.IsDefault == false {
+				vpcToBeDeleted = append(vpcToBeDeleted, vpcs.VpcId)
+			}
+		}
+
+		if vpcList.NextToken != nil {
+			token = vpcList.NextToken
+		} else {
+			break
+		}
+	}
+
+	return vpcToBeDeleted, nil
+}
+
+func CleanVpcInstances(client clientpkg.Client) {
+	vpcToBeDeleted, err := ListVPCforDeletion(client)
+	if err != nil {
+		fmt.Println("cant list VPC")
+		return
+	}
+	fmt.Println(vpcToBeDeleted)
+	_ = DeleteVpcInstances(client, vpcToBeDeleted)
+
+	// need to clear VPN connection now , VPC gateway has been detached already by now (function call in -> deleteVpcInstances()) , if not this will throw an error .
+	_ = DeleteVpnConnections(client)
+}
+
+func DeleteVpcInstances(client clientpkg.Client, vpcToBeDeleted []*string) error {
+
+	errFlag := false
+	for _, vpcId := range vpcToBeDeleted {
+
+		//need to clean out the dependencies
+		// EC2 + S3 are already cleaned out by now
+		// delete vpc endpoints
+		err := DeleteVpcEndpoint(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// clear out all ELB
+		err = DeleteELB(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// clear out all network load balancer
+		err = DeleteNetworkLoadBalancer(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// delete NAT gateway
+		err = DeleteNatgateway(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// detach and delete network interface
+		err = DetachAndDeleteNetworkInterface(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//detach any internet gateway , and delete it
+		err = DeleteGateway(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//detach VPN gateway
+		err = DetachVpnGateway(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//cleaning network ACL
+		err = DeleteNetworkAcl(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//now cleaning route tables
+		err = DeleteRouteTables(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// now cleaning subnets associated with that vpc id
+		err = DeleteSubnetsForVPC(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//now cleaning security groups
+		err = DeleteSecurityGroups(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+
+		_, err = client.DeleteVpc(&ec2.DeleteVpcInput{VpcId: vpcId})
+		if err != nil {
+			errFlag = true
+			fmt.Println("ERROR deleting VPC :", *vpcId)
+			fmt.Println(err)
+		}
+	}
+
+	if errFlag == true {
+		return ErrVpcNotDelete
+	}
+	fmt.Println("VPC's deleted successfully for this region  ")
+	return nil
+
+}
+
+func DeleteELB(client clientpkg.Client, vpcID *string) error {
+
+	errFlag := false
+	var marker *string
+	var errr = errors.New("ERROR")
```

Comment:
What is this for? The name isn't very descriptive and I'm curious why a new error needs to be created